astropy/table/table.py

```diff
@@ -842,8 +842,8 @@ def __getitem__(self, item):
             return self.columns[item]
         elif isinstance(item, (int, np.integer)):
             return self.Row(self, item)
-        elif isinstance(item, (tuple, list)) and all(isinstance(x, six.string_types)
-                                                     for x in item):
+        elif (isinstance(item, (tuple, list)) and item and
```

Comment:
ISTM just `(tuple, list)` here is a bit limited.  But since that was already the case, and and going to be the most typical case, and this otherwise resolves the issue, I don't see any reason to do anything about it right now.